src/iface/interface/mod.rs

```diff
@@ -2129,12 +2129,20 @@ impl<'a> InterfaceInner<'a> {
 
         // If the medium is Ethernet, then we need to retrieve the destination hardware address.
         #[cfg(feature = "medium-ethernet")]
-        let (dst_hardware_addr, tx_token) =
-            match self.lookup_hardware_addr(tx_token, &ip_repr.src_addr(), &ip_repr.dst_addr())? {
+        let mut dst_hardware_addr = EthernetAddress([0, 0, 0, 0, 0, 0]);
```

Comment:
Could you implement this with a `match` instead of mutating variables? 

`let (dst_hardware_addr, tx_token) = match self.caps.medium { ... }